elections.js

```diff
@@ -6,41 +6,68 @@ const virtualChainId = 1100000;
+        const responseProcessingStarted = await client.sendQuery(queryProcessingStarted);
 
-    const _isProcessingPeriod = await client.createQuery(orbsVotingContractName, "isProcessingPeriod", []);
-    const responseIsProcessingPeriod = await client.sendQuery(_isProcessingPeriod);
+        const _isProcessingPeriod = await client.createQuery(orbsVotingContractName, "isProcessingPeriod", []);
+        const responseIsProcessingPeriod = await client.sendQuery(_isProcessingPeriod);
 
-    const getCurrentEthereumBlockNumberQuery = await client.createQuery(orbsVotingContractName, "getCurrentEthereumBlockNumber", []);
-    const responsegetCurrentEthereumBlockNumber = await client.sendQuery(getCurrentEthereumBlockNumberQuery);
+        const getCurrentEthereumBlockNumberQuery = await client.createQuery(orbsVotingContractName, "getCurrentEthereumBlockNumber", []);
+        const responsegetCurrentEthereumBlockNumber = await client.sendQuery(getCurrentEthereumBlockNumberQuery);
 
-    const ethereumBlockNumberPerOrbs = Number(responsegetCurrentEthereumBlockNumber.outputArguments[0].value);
-    const isProcessingPeriod = Number(responseIsProcessingPeriod.outputArguments[0].value);
-    const inProgress = !(Number(responseProcessingStarted.outputArguments[0].value) === 0);
+        const ethereumBlockNumberPerOrbs = Number(responsegetCurrentEthereumBlockNumber.outputArguments[0].value);
+        const isProcessingPeriod = Number(responseIsProcessingPeriod.outputArguments[0].value);
+        const inProgress = !(Number(responseProcessingStarted.outputArguments[0].value) === 0);
 
-    const ethResp = await fetch('http://eth4.orbs.com:8080');
-    const eth = await ethResp.json();
+        const ethResp = await fetch('http://eth4.orbs.com:8080');
+        const eth = await ethResp.json();
 
-    return {
-        eth,
-        elections: {
-            ethereumBlockNumberPerOrbs,
-            isProcessingPeriod,
-            next: Number(responseElectionBlock.outputArguments[0].value),
-            inProgress,
-        }
-    };
+        return {
+            ok: true,
+            eth,
+            elections: {
+                ethereumBlockNumberPerOrbs,
+                isProcessingPeriod,
+                next: Number(responseElectionBlock.outputArguments[0].value),
+                inProgress,
+            }
+        };
+    } catch (err) {
+        console.error(`There has been a problem getting the latest elections status due to an error: ${JSON.stringify(err)}`);
+        return {
+            ok: false,
+            err,
+        };
+    }
 }
 
-(async () => {
-    console.log(await getElectionsStatus());
-})();
+const poller = async () => {
+    let result = await getServerElections();
+    if (result.ok) {
+        currentElectionsData = Object.assign({}, result);
+    }
+};
+
+let pid = setInterval(poller, 10 * 1000);
+poller();
+
+function getElectionsStatus() {
+    console.log('getElectionsStatus()');
+    if (!currentElectionsData) {
```

Comment:
for the first time I would block here - because we normally have automatic and unexpected restarts in Heroku - I don't want a random error returned to the end user if the UI is kept open while a server restarts.

On this same note - it's not a part of this PR but lets discuss why we don't have 2 dynos running at all times.

Reply:
but this would mean we should add this to the client in this PR.. and imho spreading this logic between the client and server complicates their contract.

I understand the "ugliness" of two flows. but I think there are elegant solutions. one is my proposed `await` above (maybe some gotcha that I don't notice? - i think if an exception is thrown during startup the server will quit and be restarted...) and another is storing a promise to the record instead of the record itself in `currentElectionsData` - but this is a little tricker to implement.,.